lib/cli.rb

```diff
@@ -584,10 +584,19 @@ def self.parse_inspect_command_options(host, options)
     c.flag ["remote-user", :r], type: String, required: false, default_value: @config.remote_user,
       desc: "Defines the user which is used to access the inspected system via SSH."\
         "This user needs sudo access on the remote machine or be root.", arg_name: "USER"
+    c.flag [:port, :p], type: Integer, required: false,
+      default_value: @config.default_ssh_port,
+      desc: "The SSH port of the remote server.",
+      arg_name: "SSH-PORT"
+    c.flag [:key, :k], type: String, required: false,
+      default_value: @config.default_ssh_key,
+      desc: "The private SSH key location what can be used to authenticate with the remote system.",
+      arg_name: "SSH-KEY"
 
     c.action do |_global_options, options, args|
       host = shift_arg(args, "HOSTNAME")
-      system = System.for(host, options["remote-user"])
+      system = System.for(host, remote_user: options["remote-user"], ssh_port: options["port"],
+        private_key_path: options["key"])
```

Comment:
To be consistent with ssh I would rather name the parameter `ssh_identity_file`.